### PR TITLE
test: Allow integration and unit tests to run against EOL or desired python version

### PR DESCRIPTION
**Pull request by ykim-akamai**
## 📝 Description

This PR adds workflow_dispatch inputs to allow e2e tests run against EOL or desired version as an option. 
e.g. 
image

## ✔️ How to Test

Forked e2e run with EOL:
- 3.9: https://github.com/ykim-akamai/linode_api4-python/actions/runs/11635801524/job/32405918836

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

**Review by yec-akamai — COMMENTED** [review]
The change looks good! The forked run seems failed due to a S3UploadFailedError. Is it related to the python version used?

File: .github/workflows/e2e-suite.yml [hunk 1]
```diff
@@ -3,9 +3,18 @@ name: E2E Tests
 on:
   workflow_dispatch:
     inputs:
-      commit_sha:
-        description: 'The hash value of the commit.'
-        required: true
+      python-version:
```
Comment (lgarber-akamai) [inline .github/workflows/e2e-suite.yml:6]: When you get a chance, could [this fix](https://github.com/linode/linode_api4-python/pull/473/files) also be applied here?

Comment (ykim-akamai) [reply .github/workflows/e2e-suite.yml:6]: Thanks for the reminder!

**Comment by ykim-akamai** [comment]
> The change looks good! The forked run seems failed due to a S3UploadFailedError. Is it related to the python version used?

Just incremented the EOL version from 3.8 to 3.9

Forked run - https://github.com/ykim-akamai/py-metadata/actions/runs/11900598917/job/33161781299

**Review by yec-akamai — APPROVED** [review]
LGTM

**Pull request merged**